File: src/modules/stored_file/services/stored_file_download_service.py
from django.db.models import Q
from src.infra.file_system.implementations.aws_file_system import AwsFileSystem
from src.modules.stored_file.models import StoredFile
from src.shared.exceptions.not_found_exception import NotFoundException


class StoredFileDownloadService:
    @classmethod
    def execute(self, uuid: str, is_public: bool | None = None):

        if is_public is not None:
            stored_file = StoredFile.objects.filter(
                uuid=uuid, is_public=is_public
            ).first()
        else:
            stored_file = StoredFile.objects.filter(uuid=uuid).first()

        if stored_file is None:
            raise NotFoundException("Arquivo não encontrado!")

        file = AwsFileSystem.get(stored_file.relative_path, stored_file.stored_name)

        return {
            "name": file["name"],
            "relative_path": stored_file.relative_path,
            "content": file["content"],
            "content_type": stored_file.content_type,
            "length": file["length"],
        }

File: src/modules/stored-file/services/stored-file-download.service.ts
import { Injectable, NotFoundException } from '@nestjs/common';
import { PrismaService } from 'src/infra/database/prisma.service';
import { FileAdapter } from 'src/infra/file-system/adapters/file-adapter';
import { AwsFileSystem } from 'src/infra/file-system/implementations/aws.file-system';

@Injectable()
export class StoredFileDownloadService {
  constructor(
    private prisma: PrismaService,
    private awsFileSystem: AwsFileSystem,
  ) {}
  async execute(uuid: string, isPublic?: boolean): Promise<FileAdapter> {
    const storedFile = await this.prisma.storedFile.findUnique({
      where: { uuid: uuid, isPublic: isPublic },
    });

    if (!storedFile) throw new NotFoundException('Arquivo não encontrado!');

    return await this.awsFileSystem.get(
      storedFile.relativePath,
      storedFile.storedName,
    );
  }
}
